gp/genome.py

```diff
@@ -53,45 +73,46 @@ def __init__(self, n_inputs, n_outputs, n_columns, n_rows, levels_back, primitiv
             1 + self._primitives.max_arity
         )  # one function gene + multiple input genes
 
-        self._dna = None  # stores dna as list of alleles for all regions
+        self._dna: List[int] = []  # stores dna as list of alleles for all regions
 
         # constants used as identifiers for input and output nodes
-        self._id_input_node = -1
-        self._id_output_node = -2
+        self._id_input_node: int = ID_INPUT_NODE
```

Comment:
omg this is soo much better with globals :heart_eyes: but now i'm starting to think. shouldn't we just introduce `InputNode` and `OutputNode` as always-present members of `Primitives` that would hence have function id 0 and 1 respectively? The numbering for all other nodes would then start at 2. this seems more natural, doesn't it? and we would need only one magic number of unused gene (which i would prefer to call "non-coding gene" btw ;) )

Reply:
also: do we need these members if we have the globals?

Reply:
Out of scope of this PR. I created a separate ticket. https://github.com/Happy-Algorithms-League/python-gp/issues/105

Reply:
I did rename the `unused_gene` to `non_coding_gene`.